### ISSUE TITLE
Leading space stripped from cells

### ISSUE THREAD

**Issue by wwboynton** [peter-murach/tty-table#2]
Peter,
First off, excellent update. This tty-table split has solved a number of small bugs and annoyances and has made the library generally nicer to use. I'm very pleased with it!

That said, I've noticed an annoying behaviour. When printing a table, cells whose strings contain leading whitespace are stripped. This would make sense in some contexts, but when getting back information from a database it can be important to know exactly the contents of something being printed in the table. I wish there were a way to change this behaviour.

In the past, I've noticed that coloring and stripping of fields have both had slightly different quirks in multiline and non-multiline mode, but I'm not sure if that's the case anymore.

**Comment by wwboynton** [comment]
If this is any help, here is the small file of gross patches for in-table colors and leading cell spaces that I had been using before the update, which still seem to work about as well/poorly as they did before. Namely, this patch allows the leading whitespace to persist, but begins truncating fields with a '...' if I add any padding.

    module TTY
      class Text
        class Wrapping
          def wrap_line(line)
            wrap_at = actual_length line
            line.gsub(/\n/, ' ').squeeze(' ')
                .gsub(/(.{1,#{wrap_at}})(?:\s+|$\n?)|(.{1,#{wrap_at}})/, "\\1\\2\n")
          end
        end
      end
      class Table
        module Operation
          class Padding
            def pad_single_line(text)
              pad_around(text)
            end
    
            def pad_multi_line(text)
              res = text.split("\n", -1).map { |part| pad_around(part) }.join("\n")
              res
            end
          end
        end
        class Field
          def length
            pastel      = Pastel.new
            strip_lines = lines.map { |line| pastel.strip(line) }
            (strip_lines.max_by(&:length) || '').size
          end
        end
      end
    end

**Comment by peter-murach** [comment]
Hi Wesley, 
I'm generally rewriting a lot of internal stuff which includes simplifying a lot of calls and removing unnecessary bloat. I must admit I have written some less than stellar stuff and I need to refactor and improve it. 

One particular bug that I have found which will mean a lot of internal changes is to do with whitespace as you have rightly pointed out. In particular, the `verse` library is at fault. I have discovered that once I was rewriting how padding works. It matches what you have described regarding the multiline mode. The wrapping in [verse](https://github.com/peter-murach/verse) library does change whitespace due to the fact that wrapping text is generally a hard thing to do considering the possibilities. I will attempt to fix it there!

In the future, all formatting operations will be handled by `verse` dependency.
